### PR TITLE
Adding support for custom formatting of argument output (#3408)

### PR DESCRIPTION
**Pull request by unt01d**
Enables custom formatting of argument output via the creation of argument classes which implement the `Codeception\Step\Argument\FormattedOutput` interface.  The argument class `Codeception\Step\Argument\PasswordArgument` is provided as this is a common use case.

Now we can do e.g.:

```
<?php

use Codeception\Step\Argument\PasswordArgument;

class LoginCest
{
    public function testLogin(AcceptanceTester $I)
    {
        $I->amOnPage('/login');

        $I->fillField('username', 'Bob');
        $I->fillField('password', new PasswordArgument('password'));
        $I->click('Log in');

        $I->see('Hi, Bob!');
    }
}
```

Which will give us:

```
$ codecept run acceptance --steps
Codeception PHP Testing Framework v2.3.7
Powered by PHPUnit 5.7.26 by Sebastian Bergmann and contributors.

Acceptance Tests (1) -------------------------------------------------
LoginCest: Test login
Signature: LoginCest:testlLogin
Test: tests/acceptance/LoginCest.php:testLogin
Scenario --
 I am on page "/login"
 I fill field "username","Bob"
 I fill field "password","******"
 I click "Log in"
 I see "Hi, Bob!"
 PASSED 

----------------------------------------------------------------------


Time: 505 ms, Memory: 15.50MB

OK (1 test, 1 assertion)

```

### CODE REVIEW

File: tests/unit/Codeception/StepTest.php [hunk 1]
```diff
@@ -114,4 +114,14 @@ public function testSeeMultiLineStringInSingleLine()
         $output = $step->toString(200);
         $this->assertEquals('see "aaaa\nbbbb\nc"', $output);
     }
+
+    public function testFormattedOutput()
+    {
+        $argument = $this->createMock('\Codeception\Step\Argument\FormattedOutput');
```
Comment (DavertMik) [inline tests/unit/Codeception/StepTest.php]: Please use `Codeception\Util\Stub::make` instead. `$this->createMock` is not available in older PHPUnit versions. That's why tests are failing

**Comment by unt01d** [comment]
Done!  Had to use `Codeception\Util\Stub::makeEmpty` since I'm stubbing the interface.

**Comment by unt01d** [comment]
Any idea why the web test is failing @DavertMik?  Seems to work for me locally.

**Comment by DavertMik** [comment]
Did you run WebDriver tests?

```
./codecept run web
```
It says

```
1) WebDriverTest: Password argument
 Test  tests/web/WebDriverTest.php:testPasswordArgument
Failed asserting that two strings are equal.
- Expected | + Actual
@@ @@
-'thisissecret'
+''

```

**Comment by unt01d** [comment]
Ah that was it, I'd run the PhpBrowser tests only.  Added support for the WebDriver by [explicitly casting the value to a string](https://github.com/Codeception/Codeception/pull/4743/commits/b486768ffae5562b78b907001f52ab23f23c5b8c).

**Comment by unt01d** [comment]
Hopefully this is ready to go now, @DavertMik.  I think some network issue caused the AppVeyor build failure?

**Comment by ptbmt** [comment]
@DavertMik push+

**Comment by DavertMik** [comment]
Passed!

**Pull request merged**